### ISSUE TITLE
Integration tests timeout on Windows using local MySQL server

### ISSUE THREAD

**Issue by RJPercival** [google/trillian#733]
```
$ go test github.com/google/trillian/integration
E0711 11:12:55.415149   12272 log_operation_manager.go:388] failed to execute operation on logs: failed to retrieve full list of log IDs: failed to get tx for retrieving logIDs: context canceled
*** Test killed: ran too long (10m0s).
```

**Comment by RJPercival** [comment]
After seeing a similar timeout occur after reducing the max number of connections to the MySQL server (in PR #722), I tried increasing the max_connections MySQL server parameter from the default of 151 to 500. This had no effect.

**Comment by RJPercival** [comment]
The cause of this was a bug in PR #734 that resulted in sequencing passes failing, but the integration test not reporting this as an error. Instead, it would just spin attempting pass after pass until the timeout was reached. The integration test should be made to abort if a terminal error occurs.

**Comment by Martin2112** [comment]
Is this still an issue? I haven't seen this fail recently.

**Comment by RJPercival** [comment]
Yes, [the LogOperationManager swallows sequencing errors](https://github.com/google/trillian/blob/master/server/log_operation_manager.go#L344) and so the integration tests can't detect that there's a problem - they just run until the timeout. There's a relevant TODO here: https://github.com/google/trillian/blob/master/server/log_operation_manager.go#L378

**Comment by Martin2112** [comment]
The TODO seems to no longer be there. Not sure if this is fixed.

**Comment by RJPercival** [comment]
[The LogOperationManager still swallows these errors](https://github.com/google/trillian/blob/master/server/log_operation_manager.go#L408).

**Comment by Martin2112** [comment]
Fair enough.

**Comment by RJPercival** [comment]
The TODO is now here: https://github.com/google/trillian/blob/master/server/log_operation_manager.go#L442

**Comment by Martin2112** [comment]
I added error metrics in #957, doesn't help tests much but you can see signer errors on production logs.

**Comment by paulmattei** [comment]
Next step: The integration test should use the new metric

**Comment by pavelkalinnikov** [comment]
Blocked on #1640.
